emission/tests/pipelineTests/TestExpectationPipeline.py

```diff
@@ -0,0 +1,115 @@
+
+    def tearDown(self):
+        self.reset_all()
+
+    def run_pipeline(self, algorithms):
+        primary_algorithms_stash = eacilp.primary_algorithms
+        eacilp.primary_algorithms = algorithms
+        test_options_stash = eaue._test_options
+        eaue._test_options["preprocess_trip"] = lambda trip: self.preprocess(trip)
+        etc.runIntakePipeline(self.testUUID)
+        eacilp.primary_algorithms = primary_algorithms_stash
+        eaue._test_options = test_options_stash
+    
+    def preprocess(self, trip):
+        # See eacilp.placeholder_predictor_2 for an explanation of the "fingerprint" technique
+        fingerprint = trip["data"]["start_local_dt"]["hour"]*60+trip["data"]["start_local_dt"]["minute"]
+        trip["data"]["end_ts"] = self.contrived_dates[fingerprint].float_timestamp
+        trip["data"]["end_local_dt"]["timezone"] = self.tz
+
+    def reset_all(self):
+        etc.dropAllCollections(edb._get_current_db())
+    
+    # Tests that the fields from the inferred trip are carried over into the expected trip correctly
+    def testPipelineIntegrity(self):
+        self.assertGreaterEqual(len(self.expected_trips), 1)  # Make sure we've actually loaded trips
+        self.assertEqual(len(self.expected_trips), len(self.inferred_trips))
+        for expected_trip in self.expected_trips:
+            inferred_id = expected_trip["data"]["inferred_trip"]
+            self.assertIn(inferred_id, self.inferred_id_to_trip.keys())
+            inferred_trip = self.inferred_id_to_trip[inferred_id]
+            self.assertEqual(expected_trip["data"]["raw_trip"], inferred_trip["data"]["raw_trip"])
+            self.assertEqual(expected_trip["data"]["cleaned_trip"], inferred_trip["data"]["cleaned_trip"])
+            self.assertEqual(expected_trip["data"]["inferred_labels"], inferred_trip["data"]["inferred_labels"])
+    
+    def testRawAgainstAnswers(self):
+        answers = {
+            494: {"type": "randomDays", "value": 2},
+            565: {"type": "all"},
+            795: {"type": "none"},
+            805: {"type": "all"},
+            880: {"type": "all"},
+            960: {"type": "randomFraction", "value": 0.05}
+        }
+        for trip in self.expected_trips:
+            fingerprint = trip["data"]["start_local_dt"]["hour"]*60+trip["data"]["start_local_dt"]["minute"]
+            self.assertEqual(eace.get_expectation(trip), answers[fingerprint])
+
+    def testProcessedAgainstAnswers(self):
+        answers = {
+            494: None,
+            565: True,
+            795: False,
+            805: True,
+            880: True,
+            960: None
+        }
+        for trip in self.expected_trips:
+            fingerprint = trip["data"]["start_local_dt"]["hour"]*60+trip["data"]["start_local_dt"]["minute"]
+            if answers[fingerprint] is not None: self.assertEqual(trip["data"]["expectation"]["to_label"], answers[fingerprint])
```

Comment:
why do you need this? won't the `"randomDays"` expectations also return `True` with your current implementation?

Reply:
I want the tests to enforce the actual intended behavior, not the current implementation. The actual intended behavior for the random options is not currently well-defined, so we don't test it — returning `True` for the random options is an implementation detail (and a placeholder, at that).